src/render.js

```diff
@@ -41,11 +41,12 @@ export function render(vnode, parentDom) {
 		EMPTY_OBJ,
 		parentDom.namespaceURI,
 		oldVNode
-			? NULL
+			? null
 			: parentDom.firstChild
 				? slice.call(parentDom.childNodes)
-				: NULL,
+				: null,
 		commitQueue,
+		// @ts-expect-error TODO: Legitimate type mismatch that our use of `NULL` hid
 		oldVNode ? oldVNode._dom : parentDom.firstChild,
```

Comment:
This is a weird one. Above, we have this:

```js
let oldVNode = isHydrating ? NULL : parentDom._children;
```

For whatever reason, type inference has failed for `NULL`, so `oldVNode` was actually `any`. As such, this is a legitimate (internal) type mismatch in what `diff` expects as a parameter and what we provide. Will fix separately.